pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -69,8 +69,27 @@ type BindMechanism interface {
 	startDHCP(vmi *v1.VirtualMachineInstance) error
 }
 
-type podNICImpl struct {
-	handler NetworkHandler
+// Network configuration is split into two parts, or phases, each executed in a
+// different context.
+// Phase1 is run by virt-handler and heavylifts most configuration steps. It
+// also creates the tap device that will be passed by name to virt-launcher,
+// thus allowing unprivileged libvirt to consume a pre-configured device.
+// Phase2 is run by virt-launcher in the pod context and completes steps left
+// out of virt-handler. The reason to have a separate phase for virt-launcher
+// and not just have all the work done by virt-handler is because there is no
+// ready solution for DHCP server startup in virt-handler context yet. This is
+// a temporary limitation and the split is expected to go once the final gap is
+// closed.
+// Moving all configuration steps into virt-handler will also allow to
+// downgrade privileges for virt-launcher, specifically, to remove NET_ADMIN
+// capability. Future patches should address that. See:
+// https://github.com/kubevirt/kubevirt/issues/3085
```

Comment:
I seriously don't know if this comment belongs here; I'd keep it closer to the `SetupPodIfaces...`, but, never the less, I plan on removing it altogether on follow-up PRs, so.. not a deal breaker for me.